newrelic-weaver/src/test/java/com/newrelic/weave/weavepackage/WeavePackageAnnotationProcessingTest.java

```diff
@@ -0,0 +1,145 @@
+import java.util.Map;
+import java.util.Set;
+
+public class WeavePackageAnnotationProcessingTest {
+    private static final Set<String> expectedJaxRsAnnotationClasses = new HashSet<>();
+    private static final Set<String> expectedJaxRsMethodAnnotationClasses = new HashSet<>();
+    private static final Set<String> expectedSpringAnnotationClasses = new HashSet<>();
+    private static final Set<String> expectedSpringMethodAnnotationClasses = new HashSet<>();
+
+    private static final String SPRING_ANNOTATIONS_CLASS_NAME = SpringAnnotations.class.getName();
+    private static final String JAX_RS_ANNOTATIONS_CLASS_NAME = JaxRsAnnotations.class.getName();
+    private static final String JAX_RS_SUBRESOURCE_ANNOTATIONS_CLASS_NAME = JaxRsSubresourceAnnotations.class.getName();
+
+    private static final String SPRING_ANNOTATIONS_CLASS_NODE_NAME = "com/newrelic/weave/weavepackage/testclasses/SpringAnnotations";
+    private static final String JAX_RS_ANNOTATIONS_CLASS_NODE_NAME = "com/newrelic/weave/weavepackage/testclasses/JaxRsAnnotations";
+    private static final String JAX_RS_SUBRESOURCE_ANNOTATIONS_CLASS_NODE_NAME = "com/newrelic/weave/weavepackage/testclasses/JaxRsSubresourceAnnotations";
+
+    @BeforeClass
+    public static void init() {
+        // jax-rs class annotations
+        expectedJaxRsAnnotationClasses.add("javax.ws.rs.Path");
+        // jax-rs method annotations
+        expectedJaxRsMethodAnnotationClasses.add("javax.ws.rs.PUT");
+        expectedJaxRsMethodAnnotationClasses.add("javax.ws.rs.POST");
+        expectedJaxRsMethodAnnotationClasses.add("javax.ws.rs.GET");
+        expectedJaxRsMethodAnnotationClasses.add("javax.ws.rs.DELETE");
+        expectedJaxRsMethodAnnotationClasses.add("javax.ws.rs.HEAD");
+        expectedJaxRsMethodAnnotationClasses.add("javax.ws.rs.OPTIONS");
+        expectedJaxRsMethodAnnotationClasses.add("javax.ws.rs.Path");
+        expectedJaxRsMethodAnnotationClasses.add("javax.ws.rs.PATCH");
+
+        // Spring class annotations
+        expectedSpringAnnotationClasses.add("org.springframework.stereotype.Controller");
+        expectedSpringAnnotationClasses.add("org.springframework.web.bind.annotation.RestController");
+        // Spring method annotations
+        expectedSpringMethodAnnotationClasses.add("org.springframework.web.bind.annotation.RequestMapping");
+        expectedSpringMethodAnnotationClasses.add("org.springframework.web.bind.annotation.PatchMapping");
+        expectedSpringMethodAnnotationClasses.add("org.springframework.web.bind.annotation.PutMapping");
+        expectedSpringMethodAnnotationClasses.add("org.springframework.web.bind.annotation.GetMapping");
+        expectedSpringMethodAnnotationClasses.add("org.springframework.web.bind.annotation.PostMapping");
+        expectedSpringMethodAnnotationClasses.add("org.springframework.web.bind.annotation.DeleteMapping");
+    }
+
+    @Test
+    public void testJaxRsAnnotationProcessing() throws IOException {
```

Comment:
Suggestion for the test name. I think it'll make it easier for our future selves to see the difference in the test

`testJaxRsSubResourceAnnotationsProcessedFirst`

Reply:
Done, for both tests